redcap_hooks.php

```diff
@@ -79,14 +79,14 @@ function redcap_custom_verify_username($username) {
 
 
 // redcap_data_entry_form (REDCap >= 5.11.0)
-function redcap_data_entry_form($project_id, $record, $instrument, $event_id, $group_id) {
+function redcap_data_entry_form($project_id, $record, $instrument, $event_id, $group_id, $repeat_instance) {
```

Comment:
The REDCap documentation shows a default value of 1 for the `$repeat_instance` parameter. I was going to add the default after discussing with @kirenotneb, but then I read the documentation more carefully:

> (only used for projects with repeating instruments/events) The repeat instance number of the repeating event (if longitudinal) or the repeating instrument (if classic or longitudinal). Default value = 1. NOTE: This parameter was added in REDCap 7.3.4.

The "only used if" language makes me wonder if the behavior is more complex than a simple default value, so I'm no longer sure what to do here. I'm going to test a little to see what REDCap does if the project doesn't have repeating instruments.

Reply:
Let me know if you need an older version of REDCap to test with that doesn't support repeating constructs.

Reply:
Will do. I'm a little more worried that it will pass null in some cases and 1 in others, which would be annoying to implement.

Reply:
Okay, REDCap *does* default `$repeat_instance` to 1 in projects without repeatable instruments enabled. Setting the default value should be redundant but harmless.